tools/perftests_ci/perftest/html.py

```diff
@@ -0,0 +1,188 @@
+'''
+Copyright (c) 2020 ETH Zurich
+
+SPDX-License-Identifier: BSL-1.0
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+'''
+import contextlib
+import os
+import pathlib
+from lxml import etree as et
+
+from pyutils import log
+
+_CSS = '''
+    /* general style */
+    body {
+        font-family: sans-serif;
+        display: flex;
+        flex-wrap: wrap;
+    }
+    section {
+        flex: 0 0 auto;
+        align-items: center;
+        box-sizing: border-box;
+        padding: 20px;
+    }
+    h1 {
+        width: 100%;
+        padding: 20px;
+    }
+    /* table style */
+    table {
+        padding-bottom: 5em;
+        border-collapse: collapse;
+        table-layout:fixed;
+        max-width: calc(100vw - 60px);
+    }
+    th {
+        text-align: left;
+        border-bottom: 1px solid black;
+        padding: 0.5em;
+        word-wrap:break-word;
+    }
+    td {
+        padding: 0.5em;
+        word-wrap:break-word;
+    }
+    /* styles for good/bad/unknown entries in table */
+    .good {
+        color: #81b113;
+        background: #dfff79;
+        font-weight: bold;
+    }
+    .bad {
+        color: #c23424;
+        background: #ffd0ac;
+        font-weight: bold;
+    }
+    .unknown {
+        color: #1f65c2;
+        background: #d5ffff;
+        font-weight: bold;
+    }
+    /* image-grid style */
+    .grid-section {
+        width: 100%;
+    }
+    .grid-container {
+        width: 100%;
+        display: grid;
+        grid-template-columns: repeat(auto-fit, minmax(500px, 1fr));
+    }
+    img {
+        width: 100%;
```

Comment:
Purely subjective, but I would set a `max-width` on this:
```suggestion
        width: 100%;
        max-width: 50em;
```
or something like that. Without it the plots end up huge if one has the browser maximized.